azure-communication-ui/demo-app/src/calling/java/com/azure/android/communication/ui/callingcompositedemoapp/CallLauncherActivity.kt

```diff
@@ -238,49 +254,134 @@ class CallLauncherActivity : AppCompatActivity() {
             }
 
             if (BuildConfig.DEBUG) {
-                versionText.text = "${BuildConfig.VERSION_NAME}"
+                versionText.text = BuildConfig.VERSION_NAME
             } else {
-                versionText.text = "${BuildConfig.VERSION_NAME} (${BuildConfig.VERSION_CODE})"
+                versionText.text = "${BuildConfig.VERSION_NAME} ${BuildConfig.VERSION_CODE}"
             }
         }
 
-        handlePushNotificationAction()
+        handlePushNotificationAction(intent)
+        registerReceiver(callLauncherBroadCastReceiver, IntentFilter(CALL_LAUNCHER_BROADCAST_ACTION))
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        isActivityRunning = false
+        EndCompositeButtonView.get(this).hide()
+        EndCompositeButtonView.buttonView = null
+        unregisterReceiver(callLauncherBroadCastReceiver)
+        callCompositeManager?.destroy()
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            (application as CallLauncherApplication).telecomConnectionServiceListener?.cleanup()
+            (application as CallLauncherApplication).telecomConnectionServiceListener = null
+        }
+        (application as CallLauncherApplication).callCompositeManager = null
     }
 
     override fun onNewIntent(intent: Intent?) {
         super.onNewIntent(intent)
-        handlePushNotificationAction()
+        handlePushNotificationAction(intent!!)
     }
 
-    private fun handlePushNotificationAction() {
-        if (intent.action != null) {
-            callLauncherViewModel.handleIncomingCall(this)
-            when (intent.action) {
-                "incoming_call" -> {
-                    binding.incomingCallLayout.visibility = View.VISIBLE
-                }
-                "answer" -> {
-                    binding.incomingCallLayout.visibility = View.GONE
-                    callLauncherViewModel.acceptIncomingCall(this@CallLauncherActivity)
-                }
-                "decline" -> {
-                    binding.incomingCallLayout.visibility = View.GONE
-                    CallCompositeManager.getInstance().declineIncomingCall()
-                }
+    private fun initCallCompositeManager() {
+        if (callCompositeManager != null) {
+            return
+        }
+
+        val application = application as CallLauncherApplication
+
+        if (application.callCompositeManager != null) {
+            callCompositeManager = application.callCompositeManager
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && callCompositeManager?.isTelecomConnectionManagerInitialized() == false) {
+                callCompositeManager?.telecomConnectionManager(TelecomConnectionManager(this@CallLauncherActivity, PHONE_ACCOUNT_ID))
+            }
+            return
+        }
+
+        callCompositeManager = CallCompositeManager(this@CallLauncherActivity)
+        application.callCompositeManager = callCompositeManager
```

Comment:
leaking an activity, if CallLauncherActivity is going into Application state, it's leaked if we store any instances.